src/core/algorithms/ucc/ucc_verifier/ucc_verifier.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <vector>
+
+#include "algorithms/algorithm.h"
+#include "config/equal_nulls/type.h"
+#include "config/indices/type.h"
+#include "config/tabular_data/input_table_type.h"
+#include "model/table/column_layout_relation_data.h"
+
+namespace algos {
+
+/* Algorithm used for verify that a set of columns is UCC and retrieving useful information in
+ * case it is not */
+
+class UCCVerifier : public Algorithm {
+private:
+    /* input options */
+    config::IndicesType column_indices_;
+    config::EqNullsType is_null_equal_null_{};
+
+    config::InputTable input_table_;
+    std::shared_ptr<ColumnLayoutRelationData> relation_;
+
+    /* results of work */
+    size_t num_rows_not_ucc_ = 0;
+    std::vector<model::PLI::Cluster> clusters_not_ucc_;
+
+    void VerifyUCC();
+    void CalculateStatistics(std::deque<model::PLI::Cluster> const& clusters);
+    void RegisterOptions();
+    void LoadDataInternal() override;
+    void ResetState() final {
+        clusters_not_ucc_.clear();
+        num_rows_not_ucc_ = 0;
+    }
+
+protected:
+    void MakeExecuteOptsAvailable() override;
+    unsigned long long ExecuteInternal() override;
+
+public:
+    /* Returns true if UCC holds and false otherwise */
+    bool UCCHolds() const {
+        return clusters_not_ucc_.empty();
+    }
+
+    /* Returns the number of clusters where UCC is violated, that is, in which rows equal in the
+     * specified columns */
+    size_t GetNumClustersNotUCC() const {
+        return clusters_not_ucc_.size();
+    }
+
+    /* Returns the total number of table rows that violate the UCC */
+    size_t GetNumRowsNotUCC() const {
+        return num_rows_not_ucc_;
+    }
+
+    /* Returns clusters where UCC is violated, that is, in which rows equal in the
+     * specified columns */
+    std::vector<model::PLI::Cluster> const& GetClustersNotUCC() const {
```

Comment:
Question about method description: maybe "... in the specified columns" (instead of  "... in the UCC columns) is better because specified columns (i. e. input columns) does not hold UCC in case when returned vector is not empty. So ".. the UCC columns" is a little confusing